ivadomed/loader/bids_dataframe.py

```diff
@@ -86,25 +87,25 @@ def create_bids_dataframe(self):
             ext_ct = ('.nii.gz', '.nii')
             suffix_ct = ('ct', 'CT')
             force_index = []
-            for root, dirs, files in os.walk(path_data):
-                for file in files:
+            for path_object in path_data.glob('**/*'):
+                if path_object.is_file():
                     # Microscopy
-                    if file == "samples.tsv" or file == "samples.json":
-                        force_index.append(file)
-                    if (file.endswith(ext_microscopy) and os.path.basename(root) == "microscopy" and
-                            (root.replace(path_data, '').startswith("sub"))):
-                        force_index.append(os.path.join(root.replace(path_data, '')))
+                    if path_object.name == "samples.tsv" or path_object.name == "samples.json":
+                        force_index.append(path_object.name)
+                    if (path_object.name.endswith(ext_microscopy) and path_object.parent.name == "microscopy" and
+                            (path_object.parts[len(path_data.parts)]).startswith('sub')):
+                        force_index.append(str(Path(*path_object.parent.parts[len(path_data.parts):])))
                     # CT-scan
-                    if (file.endswith(ext_ct) and file.split('.')[0].endswith(suffix_ct) and
-                            (os.path.basename(root) == "anat" or os.path.basename(root) == "ct") and
-                            (root.replace(path_data, '').startswith("sub"))):
-                        force_index.append(os.path.join(root.replace(path_data, '')))
+                    if (path_object.name.endswith(ext_ct) and path_object.name.split('.')[0].endswith(suffix_ct) and
+                            (path_object.parent.name == "anat" or path_object.parent.name == "ct") and
+                            (path_object.parts[len(path_data.parts)]).startswith('sub')):
```

Comment:
`(path_object.parts[len(path_data.parts)]).startswith('sub'))`

Now this is the second time I see this complex name. Since they are across multiple `IF` conditions, I think we should pre-set this variable and make it very readable/easy to follow especially used in multiple places.